waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/TunnelingMetastoreClientBuilder.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (C) 2016-2017 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.waggledance.client;
+
+import static com.hotels.bdp.waggledance.client.MetaStoreClientFactory.INTERFACES;
+import static com.hotels.bdp.waggledance.client.TunnelConnectionManagerFactory.FIRST_AVAILABLE_PORT;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+
+import com.pastdev.jsch.tunnel.TunnelConnectionManager;
+
+public class TunnelingMetastoreClientBuilder {
+
+  private HiveConf hiveConf;
+  private String name;
+  private Integer reconnectionRetries;
+  private TunnelConnectionManagerFactory tunnelConnectionManagerFactory;
+  private String remoteHost;
+  private String sshRoute;
+  private String localHost;
+  private Integer remotePort;
+
+  public CloseableThriftHiveMetastoreIface build() {
+    TunnelConnectionManager tunnelConnectionManager = tunnelConnectionManagerFactory.create(sshRoute, localHost,
+        FIRST_AVAILABLE_PORT, remoteHost, remotePort);
+    TunnelConfiguration tunnelConfiguration = new TunnelConfiguration(hiveConf, tunnelConnectionManager, sshRoute, localHost,
+        remoteHost, remotePort);
+    HiveConf localHiveConf = tunnelConfiguration.getConf();
+    return clientFromLocalHiveConf(tunnelConnectionManager, localHiveConf);
+  }
+
+  private CloseableThriftHiveMetastoreIface clientFromLocalHiveConf(
+      TunnelConnectionManager tunnelConnectionManager,
+      HiveConf localHiveConf) {
+    CloseableThriftHiveMetastoreIface client = new MetaStoreClientFactory().newInstance(localHiveConf, name,
```

Comment:
remove `new MetaStoreClientFactory()` and add a method to set this in the builder